lib/spack/spack/solver/asp.py

```diff
@@ -1338,6 +1352,26 @@ def _rule_from_str(
             message=None,
         )
 
+    def reject_requirement_constraint(
+        self, pkg_name: str, *, constraint: str, kind: RequirementKind
+    ) -> bool:
+        """Returns True if a requirement constraint should be rejected"""
+        if kind == RequirementKind.DEFAULT:
```

Comment:
Maybe never call this if `kind != DEFAULT` and drop the `kind` argument?

Reply:
Hmm, I thought about that, but then I'll just have to push the check on `kind` to the caller in three places, and disconnect it from the check that the variants, compilers, versions etc. exists.

If you worry about a change in API I can add a leading `_`.